pkg/datapath/loader/cache.go

```diff
@@ -186,15 +189,15 @@ func (o *ObjectCache) build(ctx context.Context, cfg *templateCfg, hash string)
 //
 // Returns the path to the compiled template datapath object and whether the
 // object was compiled, or an error.
-func (o *ObjectCache) FetchOrCompile(ctx context.Context, cfg datapath.EndpointConfiguration) (string, bool, error) {
+func (o *ObjectCache) FetchOrCompile(ctx context.Context, cfg datapath.EndpointConfiguration, stats *SpanStat) (string, bool, error) {
```

Comment:
Could we gain anything if `SpanStat` was an interface?

Reply:
I don't think so, there's only one user of this and we don't want to inject dependencies here. Earlier this was being called from out-of-package, but not any more. This can now be hidden.